python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+        y -= np.mean(y)
+        y = y[isBg]
+        b = np.ones_like(x, dtype=np.float64)
+
+        M = np.vstack([b]).T  # order = 0
+        if order == 1:
+            M = np.vstack([b, x, y]).T
+        elif order == 2:
+            M = np.vstack([b, x, y, x**2., y**2., x*y]).T
+
+        pars = np.linalg.lstsq(M, B)[0]
+        return pars
+
+    def genStarModel(self, bbox, psf, xcen, ycen, flux):
+        """Generate model (2-d array) of a 'star' (single PSF) centered at given coordinates
+
+        @param bbox Bounding box marking pixel coordinates for generated model
+        @param psf Psf model used to generate the 'star'
+        @param xcen Desired x-centroid of the 'star'
+        @param ycen Desired y-centroid of the 'star'
+        @param flux Desired flux of the 'star'
+
+        @return 2-d numpy.array of width/height matching input bbox,
+        containing PSF with given centroid and flux
+        """
+
+        # Generate the psf image, normalize to flux
+        psf_img = psf.computeImage(afwGeom.Point2D(xcen, ycen)).convertF()
+        psf_img_sum = np.nansum(psf_img.getArray())
+        psf_img *= (flux/psf_img_sum)
+
+        # Clip the PSF image bounding box to fall within the footprint bounding box
+        psf_box = psf_img.getBBox()
+        psf_box.clip(bbox)
+        psf_img = afwImage.ImageF(psf_img, psf_box, afwImage.PARENT)
+
+        # Then actually crop the psf image.
+        # Usually not necessary, but if the dipole is near the edge of the image...
+        # Would be nice if we could compare original pos_box with clipped pos_box and
+        #     see if it actually was clipped.
+        p_Im = afwImage.ImageF(bbox)
+        tmpSubim = afwImage.ImageF(p_Im, psf_box, afwImage.PARENT)
+        tmpSubim += psf_img
+
+        return p_Im
+
+    @staticmethod
+    def genDipoleModel(x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=None,
+                       b=None, x1=None, y1=None, xy=None, x2=None, y2=None,
+                       bNeg=None, x1Neg=None, y1Neg=None, xyNeg=None, x2Neg=None, y2Neg=None,
+                       **kwargs):
+        """Generate dipole model with given parameters.
+
+        This is the functor whose sum-of-squared difference from data
+        is minimized by `lmfit`. Thus it must be static. However, it
+        just defers to `self.genDipoleModelImpl()`, where `self` comes
+        out of kwargs['myself'].
+
+        @see genDipoleModelImpl
+        """
+        algObject = kwargs.pop('algObject')
+        return algObject.genDipoleModelImpl(x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=fluxNeg,
+                                            b=b, x1=x1, y1=y1, xy=xy, x2=x2, y2=y2,
+                                            bNeg=bNeg, x1Neg=x1Neg, y1Neg=y1Neg, xyNeg=xyNeg,
+                                            x2Neg=x2Neg, y2Neg=y2Neg, **kwargs)
+
+    def genDipoleModelImpl(self, x, flux, xcenPos, ycenPos, xcenNeg, ycenNeg, fluxNeg=None,
```

Comment:
Consider renaming this to __call__, thus making the class a true functor. That eliminates the need for the static method  `genDipoleModel`. If you do that, it does argue in favor of retaining the implementation as a class.

This suggestion only makes sense this method is the only functor-like use of this class.